### PR TITLE
Issue #35: satelite projects should use latest checkstyle as soon at is appears in maven repo

### LINKED ISSUE #35: satelite projects should use latest checkstyle as soon at is appears in maven repo
Right now we do bump a version in a few projects https://github.com/checkstyle/checkstyle/wiki/How-to-make-a-release#update-for-released-version-reference right after release.

Let's find a way to define in pom.xml the latest snapshot and most recent version. And the latest released version when we define dependencies. To let us skip such updates at all.

CI of such projects should be able to get the latest version from https://repo1.maven.org/maven2/com/puppycrawl/tools/checkstyle/maven-metadata.xml and run projects with such version.

We can have in all such repos, a GitHub action that runs once a day check https://repo1.maven.org/maven2/com/puppycrawl/tools/checkstyle/maven-metadata.xml, put such values in required files, and send PRs with changes (PR is necessary as we need notice is any problems happens if we introduce breaking changes)

TASK:

- [ ] Version Bump in `ivy.xml`
- [ ] Version Bump in `pom.xml`
- [ ] Version Bump in `build.gradle`

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
Partially Fixes #35 

Testing:
```
rahul@rk7:~/Desktop/opensource/checkstyle-samples$ ./.ci/update-pom-version.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  3785  100  3785    0     0   2595      0  0:00:01  0:00:01 --:--:--  2596
LATEST_VERSION=10.3.2
MVN_CURRENT_VERSION=0.0.1-SNAPSHOT
Checking Version for: maven-project/pom.xml
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO] 
[INFO] -------------< com.github.sevntu-checkstyle:maven-project >-------------
[INFO] Building maven-project 0.0.1-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- versions-maven-plugin:2.11.0:set (default-cli) @ maven-project ---
[INFO] Searching for local aggregator root...
[INFO] Local aggregation root: /home/rahul/Desktop/opensource/checkstyle-samples/maven-project
[INFO] Processing change of com.github.sevntu-checkstyle:maven-project:0.0.1-SNAPSHOT -> 10.3.2-SNAPSHOT
[INFO] Processing com.github.sevntu-checkstyle:maven-project
[INFO]     Updating project com.github.sevntu-checkstyle:maven-project
[INFO]         from version 0.0.1-SNAPSHOT to 10.3.2-SNAPSHOT
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.328 s
[INFO] Finished at: 2022-08-19T16:21:01+05:30
[INFO] ------------------------------------------------------------------------
Version updated to 10.3.2-SNAPSHOT
rahul@rk7:~/Desktop/opensource/checkstyle-samples$ 
```

### CODE REVIEW

File: .ci/bump-maven-pom-version.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+set -e
```
Comment (romani) [inline .ci/bump-maven-pom-version.sh:2]: apply comments from https://github.com/checkstyle/contribution/pull/635#pullrequestreview-1079599409

File: .ci/bump-maven-pom-version.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+echo LATEST_VERSION="$LATEST_VERSION"
+
+MVN_CURRENT_VERSION=$(cat $POM_FILE | grep "^    <version>.*</version>$" | awk -F'[><]' '{print $3}')
+echo MVN_CURRENT_VERSION="$MVN_CURRENT_VERSION"
+
+echo "Checking Version for: "$POM_FILE
+if [ "$MVN_CURRENT_VERSION" != "$LATEST_VERSION"  ]; then
+    cd maven-project
+    mvn -e --no-transfer-progress versions:set -DgroupId=com.github.sevntu-checkstyle -DartifactId=maven-project \
+   -DoldVersion="$MVN_CURRENT_VERSION" -DnewVersion="$LATEST_VERSION-SNAPSHOT"
```
Comment (romani) [inline .ci/bump-maven-pom-version.sh:17]: respect indentaiton of +2 for line wrap.
```
mvn -e --no-transfer-progress versions:set \
  -DgroupId=com.github.sevntu-checkstyle -DartifactId=maven-project \
  -DoldVersion="$MVN_CURRENT_VERSION" -DnewVersion="$LATEST_VERSION-SNAPSHOT"
```

**Review by romani — CHANGES_REQUESTED** [review]
item:

File: .ci/bump-maven-pom-version.sh (hunk 1 above)
Comment (Rahulkhinchi03) [reply .ci/bump-maven-pom-version.sh]: Changes Done

File: .ci/bump-maven-pom-version.sh (hunk 2 above)
Comment (Rahulkhinchi03) [reply .ci/bump-maven-pom-version.sh]: Just for confirmation,
Do I need to update the version or checkstyle version?

**Review by romani — APPROVED** [review]
script is fixed and tested.

**Pull request merged**